src/client.rs

```diff
@@ -1040,48 +1022,49 @@ impl Spotify {
     /// [Reference](https://developer.spotify.com/documentation/web-api/reference/#endpoint-get-users-saved-tracks)
     pub fn current_user_saved_tracks(&self) -> impl Paginator<ClientResult<SavedTrack>> + '_ {
         paginate(
-            move |limit, offset| self.current_user_saved_tracks_manual(limit, offset),
+            move |limit, offset| self.current_user_saved_tracks_manual(Some(limit), Some(offset)),
             self.pagination_chunks,
         )
     }
 
     /// The manually paginated version of
     /// [`Spotify::current_user_saved_tracks`].
     #[maybe_async]
-    pub async fn current_user_saved_tracks_manual<L: Into<Option<u32>>, O: Into<Option<u32>>>(
+    pub async fn current_user_saved_tracks_manual(
         &self,
-        limit: L,
-        offset: O,
+        limit: Option<u32>,
+        offset: Option<u32>,
     ) -> ClientResult<Page<SavedTrack>> {
-        let mut params = Query::with_capacity(2);
-        let limit = limit.into().unwrap_or(20).to_string();
-        let offset = offset.into().unwrap_or(0).to_string();
-        params.insert("limit", &limit);
-        params.insert("offset", &offset);
+        let limit = limit.map(|s| s.to_string());
+        let offset = offset.map(|s| s.to_string());
+        let params = build_map! {
+            optional "limit": limit.as_deref(),
+            optional "offset": offset.as_deref(),
+        };
+
         let result = self.endpoint_get("me/tracks", &params).await?;
         self.convert_result(&result)
     }
 
     /// Gets a list of the artists followed by the current authorized user.
     ///
     /// Parameters:
-    /// - limit - the number of tracks to return
     /// - after - the last artist ID retrieved from the previous request
+    /// - limit - the number of tracks to return
     ///
     /// [Reference](https://developer.spotify.com/documentation/web-api/reference/#endpoint-get-followed)
     #[maybe_async]
-    pub async fn current_user_followed_artists<L: Into<Option<u32>>>(
+    pub async fn current_user_followed_artists(
         &self,
-        limit: L,
-        after: Option<String>,
+        after: Option<&str>,
+        limit: Option<u32>,
     ) -> ClientResult<CursorBasedPage<FullArtist>> {
-        let mut params = Query::with_capacity(2);
-        let limit = limit.into().unwrap_or(20).to_string();
-        params.insert("limit", &limit);
-        params.insert("type", Type::Artist.as_ref());
-        if let Some(ref after) = after {
-            params.insert("after", &after);
-        }
+        let limit = limit.map(|s| s.to_string());
+        let params = build_map! {
+            "r#type": Type::Artist.as_ref(),
```

Comment:
It seems we haven't handle the raw identifiers, then the `r#type` will be inserted into the parameter map?

Reply:
whoops, my bad